### ISSUE TITLE
rspack/core (0.5.2) breaks ModuleFederation shared option

### ISSUE THREAD

**Issue by HEHOHADA** [web-infra-dev/rspack#5494]
### System Info

```
 System:
    OS: Linux 6.1 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (2) x64 AMD EPYC
    Memory: 1.04 GB / 2.01 GB
    Container: Yes
    Shell: Unknown
  Binaries:
    Node: 20.9.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.8.1 - /usr/local/bin/npm
```

### Details

After updating to the latest version and  use `ModuleFederationPlugin` with active option `shared`
The build doesn't work if you enable the package `react-dom`
if you remove the option `shared` or rollback to 0.5.1 version everything starts working

### Reproduce link

https://codesandbox.io/p/devbox/hardcore-monad-9mjrff?file=%2Fhome%2Frspack.config.js

### Reproduce Steps

```
cd /home
npm run build
```

**Comment by messaooudi** [comment]
We have the same issue with packages like `antd`, `dayjs`..., if they are added as shared dependencies, the build freezes at `sealing dependencies` and then the process is killed `Killed ELIFECYCLE  Command failed with exit code 137.`

**Comment by douglaszaltron** [comment]
This directly affects `rsbuild`; I'm facing the same issue there as well. With the latest updates, the shared functionality has stopped working.

https://github.com/web-infra-dev/rsbuild/issues/1234

**Comment by ahabhgk** [comment]
Fixed in https://github.com/web-infra-dev/rspack/pull/5458

**Comment by ahabhgk** [comment]
Fix is released in 0.5.3
